frontend/src/metabase/lib/hooks.js

```diff
@@ -0,0 +1,34 @@
+import { useRef, useEffect, useMemo, useState } from "react";
+
+export function useAsyncFunction(fn) {
+  const [isPending, setIsPending] = useState(false);
+  const isUnmountedRef = useRef(false);
+
+  useEffect(() => {
+    () => {
```

Comment:
You missed the `return` keyword (or removing curly braces)

Reply:
fixed